### PR TITLE
[HEVCe] Fix temporal layers reset issue

### PR DESCRIPTION
**Pull request by JeremyShang**

### CODE REVIEW

**Review by yannanba — DISMISSED** [review]

File: _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_legacy.cpp [hunk 1]
```diff
@@ -539,6 +540,20 @@ void Legacy::SetInherited(ParamInheritance& par)
         INHERIT_OPT(VideoFullRange);
         INHERIT_OPT(ColourDescriptionPresent);
     });
+
+    par.m_ebInheritDefault[MFX_EXTBUFF_AVC_TEMPORAL_LAYERS].emplace_back(
+    [](const mfxVideoParam& /*parInit*/
```
Comment (MaximIntel) [inline _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_legacy.cpp:545]: indention is not correct here.

**Review by MaximIntel — CHANGES_REQUESTED** [review]

File: _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_legacy.cpp (hunk 1 above)
Comment (JeremyShang) [reply _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_legacy.cpp]: Thank you! Fixed.

**Review by vilichev — APPROVED** [review]

File: _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_legacy.cpp [hunk 2]
```diff
@@ -539,6 +540,20 @@ void Legacy::SetInherited(ParamInheritance& par)
...
     });
+
+    par.m_ebInheritDefault[MFX_EXTBUFF_AVC_TEMPORAL_LAYERS].emplace_back(
+        [](const mfxVideoParam& /*parInit*/
+            , const mfxExtBuffer* pSrc
+            , const mfxVideoParam& /*parReset*/
+            , mfxExtBuffer* pDst)
+    {
+        INIT_EB(mfxExtAvcTemporalLayers);
+        for (mfxU32 i = 0; i < 7; i++)
```
Comment (lakulako) [inline _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_legacy.cpp:551]: Could 7 be changed to some define or at least add comment why 7.

**Review by lakulako — COMMENTED** [review]
Please also update copyright year

File: _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_legacy.cpp (hunk 2 above)
Comment (JeremyShang) [reply _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_legacy.cpp:553]: Thanks @lakulako , comments are added.

Comment (lakulako) [reply _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_legacy.cpp:553]: Fine, then may be add also something like "HW API limitation"?

Comment (JeremyShang) [reply _studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_legacy.cpp:555]: Sure. The value comes from HEVC spec, I have explained it in the comments.

**Comment by JeremyShang** [comment]
> Please also update copyright year

Thanks, done.

**Review by lakulako — APPROVED** [review]

**Review by vilichev — APPROVED** [review]

**Review by yannanba — APPROVED** [review]

**Review by dmitryermilov — APPROVED** [review]

**Pull request merged**